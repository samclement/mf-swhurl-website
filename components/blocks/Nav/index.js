import React from 'react'
import Link from '../../elements/Link'

export default () => {
  return (
    <nav>
      <Link href="/" label="Home" />
      &nbsp;|&nbsp;
      <Link href="/about" label="About" />
      &nbsp;|&nbsp;
      <Link href="/contact" label="Contact" />
    </nav>
  )
}
