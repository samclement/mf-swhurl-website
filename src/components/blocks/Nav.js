import React from 'react'
import NavLink from '../elements/NavLink'
import styled from 'styled-components'

const StyledTitle = styled.div`
  font-weight: bold;
  text-transform: capitalize;
`

const StyledNav = styled.nav`
  display: flex;
  flex-flow: row wrap;
  justify-content: center;
  align-items: center;
`

export default ({ items, gender }) => {
  return (
    <StyledNav>
      <StyledTitle>{gender}</StyledTitle>
      {items.map((item, i) => {
        const url = `/plp?url=${item.url}`
        return <NavLink key={i} href={url} as={item.url} label={item.name} />
      })}
    </StyledNav>
  )
}
