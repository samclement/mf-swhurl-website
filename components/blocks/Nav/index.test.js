/* eslint-env jest */

import React from 'react'
import { render } from 'enzyme'
import 'jest-styled-components'

import Nav from './index.js'

describe('With Snapshot Testing', () => {
  it('Nav has 3 sections', () => {
    const wrapper = render(<Nav />)
    expect(wrapper).toMatchSnapshot()
  })
})
