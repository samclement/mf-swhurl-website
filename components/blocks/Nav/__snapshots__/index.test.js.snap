// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing Nav displays items 1`] = `
.c2 {
  color: darkgray;
  padding-right: 5px;
}

.c2:hover {
  color: #F00;
}

.c2:visited {
  color: darkgray;
}

.c1 {
  font-weight: bold;
  text-transform: capitalize;
  padding-right: 10px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
}

<nav
  class="c0"
>
  <div
    class="c1"
  >
    mens
  </div>
  <a
    class="c2"
    href="/mens/shop/accessories"
  >
    Accessories
  </a>
  <a
    class="c2"
    href="/mens/shop/shoes"
  >
    Shoes
  </a>
  <a
    class="c2"
    href="/mens/shop/clothing"
  >
    Clothing
  </a>
  <a
    class="c2"
    href="/mens/shop/bags"
  >
    Bags
  </a>
  <a
    class="c2"
    href="/mens/shop/homeware"
  >
    Homeware
  </a>
</nav>
`;
